Sources/NIO/Channel.swift

```diff
@@ -352,6 +352,9 @@ public enum MulticastError: Error {
     /// An attempt was made to join a multicast group that does not correspond to a multicast
     /// address.
     case illegalMulticastAddress(SocketAddress)
+
+    /// Multicast is not supported on Interface
+    case multicastNotSupported(NIONetworkInterface)
```

Comment:
If you'd like this patch to ship sooner rather than later, you can't add new errors to existing `enum`s. This is because adding cases to enumerations is a source-breaking change in Swift, as all `switch` statements are exhaustive. You'll need to add a new `enum` with a single case, I'm afraid.

Reply:
I feel that providing back the Interface in the error is more expressive than just a generic error saying multicastNotSupportedOnInterface.  

however I could also argue that one can infer which interface it was as they had to pass it in.